cli/dcoscli/node/main.py

```diff
@@ -472,9 +472,21 @@ def _list(json_, extra_field_names):
     """
 
     client = mesos.DCOSClient()
+    masters = mesos.MesosDNSClient().hosts('master.mesos.')
+    master_state = client.get_master_state()
     slaves = client.get_state_summary()['slaves']
+    for master in masters:
+        if master['ip'] == master_state['hostname']:
+            master['type'] = 'master (leader)'
+            for key in ('id', 'pid', 'version'):
+                master[key] = master_state.get(key)
```

Comment:
Read this as only adding if `extra_field_names` had those values. please ignore :)